### PR TITLE
contrail logs

### PR DESCRIPTION
**Pull request by anshprat**
The actual changes behind https://github.com/JioCloud/puppet-rjil/pull/602

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
 group :development, :test do
-  gem 'bodeco_module_helper', :git => 'https://github.com/bodeco/bodeco_module_helper.git'
-  gem 'mocha', '~> 0.12.10'
+  gem 'puppetlabs_spec_helper', :ref => '~>0.10'
+  gem 'bodeco_module_helper', :git => 'https://github.com/bodeco/bodeco_module_helper.git', :ref => 'a0de65d'
+  gem 'rspec-puppet', '~>1.0'
+  gem 'hiera-puppet-helper', :git => 'https://github.com/bobtfish/hiera-puppet-helper', :ref => '80b597aec1a'
+  gem 'rspec-puppet-utils'
+  gem 'mocha', '~>0.10.5'
+  gem 'librarian-puppet-simple'
```
Comment (bodepd) [inline Gemfile]: is this change actually required for this patch to pass?

File: files/ifmap-log4j.properties [hunk 2]
```diff
@@ -0,0 +1,10 @@
+log4j.rootLogger=INFO
+log4j.error
+
+log4j.logger.de.fhhannover.inform.irond.proc=INFO, A1
```
Comment (bodepd) [inline files/ifmap-log4j.properties]: shouldn't we templatize this so that it can be configurable?

File: templates/contrail-analytics-api.conf.erb [hunk 3]
```diff
@@ -4,8 +4,9 @@ collectors = <%= @collector_ip %>:8086
 http_server_port = 8090
 rest_api_port = 8081
 rest_api_ip = 0.0.0.0
-log_local = 0
-log_level = SYS_DEBUG
+log_file_size = 10737418240 # 10 GB
+log_level = SYS_INFO
```
Comment (bodepd) [inline templates/contrail-analytics-api.conf.erb]: this should be configurable

File: templates/contrail-api.conf.erb [hunk 4]
```diff
@@ -8,7 +8,10 @@ listen_ip_addr=<%= @api_listen %>
 listen_port=<%= @api_local_listen_port %>
 auth=keystone
 multi_tenancy=<%= @multi_tenancy %>
-log_file=/var/log/contrail/api.log
+log_file=/var/log/contrail/contrail-api.log
+log_file_size= 10737418240 # 10 GB
+log_level=SYS_INFO
```
Comment (bodepd) [inline templates/contrail-api.conf.erb]: should be configurable

File: templates/contrail-collector.conf.erb [hunk 5]
```diff
@@ -13,7 +13,10 @@
   http_server_port=8089
 # log_category=
 # log_disable=0
-  log_file=/var/log/contrail/collector.log
+  log_file=/var/log/contrail/contrail-collector.log
+  log_file_size= 10737418240 # 10 GB
+  log_level=SYS_INFO
```
Comment (bodepd) [inline templates/contrail-collector.conf.erb]: should be configurable

File: templates/contrail-control.conf.erb [hunk 6]
```diff
@@ -18,10 +18,10 @@
 # log_category=
 # log_disable=0
   log_file=/var/log/contrail/contrail-control.log
+  log_file_size= 10737418240 # 10 GB
+  log_level=SYS_INFO
```
Comment (bodepd) [inline templates/contrail-control.conf.erb]: should be configurable

File: Gemfile (hunk 1 above)
Comment (anshprat) [inline Gemfile]: Not really, this was done the other day while debugging travis exit 1. Whats better? This or the earlier version?

Comment (bodepd) [inline Gemfile]: I'd rather not conflate this into the patch if we don't need it

File: templates/discovery.conf.erb [hunk 7]
```diff
@@ -11,7 +11,9 @@ zk_server_port=<%= @zk_port %>
 listen_ip_addr=<%= @discovery_listen %>
 listen_port=<%= @discovery_local_listen_port %>
 log_local=True
-log_file=/var/log/contrail/discovery.log
+log_file=/var/log/contrail/contrail-discovery.log
```
Comment (bodepd) [inline templates/discovery.conf.erb]: shouldn't this get parameterized?

Comment (anshprat) [inline templates/discovery.conf.erb]: fixed.

File: Gemfile (hunk 1 above)
Comment (anshprat) [inline Gemfile]: done. tests passed in travis. will check my local env.

**Comment by anshprat** [comment]
@hkumar @soren @bodepd This is ready for review

File: templates/ifmap-log4j.properties.erb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+log4j.rootLogger=INFO
```
Comment (hkumarmk) [inline templates/ifmap-log4j.properties.erb]: I see this is a static file, so please move this under files and call it using source =>

File: manifests/ifmap.pp [hunk 9]
```diff
@@ -55,6 +55,13 @@
     notify  => Service['ifmap-server'],
   }
 
+  file { '/etc/ifmap-server/log4j.properties' :
+    ensure  => present,
+    require => Package['ifmap-server'],
+    content => template("${module_name}/ifmap-log4j.properties.erb"),
```
Comment (hkumarmk) [inline manifests/ifmap.pp]: I see this file just have a static content, so no need to call with template, just move that file under files/ directory and call with source param rather than content.

File: templates/ifmap-log4j.properties.erb (hunk 8 above)
Comment (anshprat) [inline templates/ifmap-log4j.properties.erb]: added missing template part

**Comment by bodepd** [comment]
needs to be rebased

**Comment by anshprat** [comment]
@bodepd rebase done with the last merge to your lint warning fixes as the base. Basically to keep the merge as part of PR history. Is that ok or should I rebase to the first commit of this PR?

File: manifests/ifmap.pp (hunk 9 above)
Comment (anshprat) [inline manifests/ifmap.pp]: Made the file configurable.

**Comment by bodepd** [comment]
the workflow for rebasing is the following:

```
git fetch origin
git rebase origin/master
# git rebase --continue as required to fix merge conflicts
# git push --force as your will have to rewrite history when you finish
```
